### PR TITLE
[otci] add diag commands support to otci

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

**Review by abtink — APPROVED** [review]
👍

File: tools/otci/otci/otci.py [hunk 1]
```diff
@@ -2344,6 +2344,198 @@ def coap_test_set_resource_content(self, content: str):
     # TODO: coap parameters <type> ["default"|<ack_timeout> <ack_random_factor_numerator> <ack_random_factor_denominator> <max_retransmit>]
     # TODO: CoAP Secure utilities
 
+    #
+    # Diag Utilities
+    #
+
```
Comment (superwhd) [inline tools/otci/otci/otci.py:2350]:
```suggestion
```
nit

File: tools/otci/otci/otci.py [hunk 2]
```diff
@@ -2344,6 +2344,198 @@ def coap_test_set_resource_content(self, content: str):
...
+    def diag_repeat_stop(self):
+        """Stop repeated packet transmission."""
+        self.execute_command('diag repeat stop')
+
+    def diag_radio_sleep(self):
+        """Enter radio sleep mode."""
+        self.execute_command('diag radio sleep')
+
+    def diag_radio_receive(self):
+        """Set radio from sleep mode to receive mode."""
```
Comment (superwhd) [inline tools/otci/otci/otci.py:2417]:
```suggestion
        """Set radio to receive mode."""
```
Is it guaranteed the previous mode is sleep mode?

**Review by superwhd — APPROVED** [review]
LGTM overall with minor questions

**Review by jwhui — APPROVED** [review]

**Pull request merged**